### PR TITLE
chore: add pry to debug unit tests

### PR DESCRIPTION
**Pull request by SomalianIvan**

### CODE REVIEW

File: sdk/Gemfile [hunk 1]
```diff
@@ -14,3 +14,7 @@ gem 'opentelemetry-common', path: '../common'
 gem 'opentelemetry-exporter-zipkin', path: '../exporter/zipkin'
 gem 'opentelemetry-instrumentation-base', path: '../instrumentation/base'
 gem 'opentelemetry-semantic_conventions', path: '../semantic_conventions'
+
+group :development, :test do
+  gem 'pry'
```
Comment (arielvalentin) [inline sdk/Gemfile:19]: @open-telemetry/ruby-approvers I see that we have a mix of uses of `pry` and `pry-byebug`. 

https://github.com/open-telemetry/opentelemetry-ruby/search?q=pry

Is there a reason for us not to include it in the SDK? 

Also if we want to add this to the SDK should this be a development dependency in the Gemspec instead?

Comment (fbogsany) [reply sdk/Gemfile:19]: I think a dev dependency in the Gemspec makes sense.

Comment (arielvalentin) [reply sdk/Gemfile:19]: Ok so I think we need to create a separate issue for those instances. I would ask @SomalianIvan for this PR that you kindly use the gemspec to declare this as a development dependency as well as add `pry-byebug`.

Comment (SomalianIvan) [reply sdk/Gemfile]: thanks for speedy reviews :)

Comment (arielvalentin) [reply sdk/Gemfile]: 🙇🏼 thank you for your contributions!

Comment (SomalianIvan) [reply sdk/Gemfile]:
```
❯ ag "pry" ./**/*.gemspec --print-all-files                                                                                                                                                                                                                                  ~/src/github.com/open-telemetry/opentelemetry-ruby
api/opentelemetry-api.gemspec

common/opentelemetry-common.gemspec

exporter/jaeger/opentelemetry-exporter-jaeger.gemspec

exporter/otlp/opentelemetry-exporter-otlp.gemspec

exporter/zipkin/opentelemetry-exporter-zipkin.gemspec

instrumentation/action_view/opentelemetry-instrumentation-action_view.gemspec

instrumentation/active_job/opentelemetry-instrumentation-active_job.gemspec
36:  spec.add_development_dependency 'pry'

instrumentation/active_model_serializers/opentelemetry-instrumentation-active_model_serializers.gemspec

instrumentation/active_record/opentelemetry-instrumentation-active_record.gemspec

instrumentation/all/opentelemetry-instrumentation-all.gemspec

instrumentation/base/opentelemetry-instrumentation-base.gemspec

instrumentation/bunny/opentelemetry-instrumentation-bunny.gemspec

instrumentation/concurrent_ruby/opentelemetry-instrumentation-concurrent_ruby.gemspec

instrumentation/dalli/opentelemetry-instrumentation-dalli.gemspec

instrumentation/delayed_job/opentelemetry-instrumentation-delayed_job.gemspec

instrumentation/ethon/opentelemetry-instrumentation-ethon.gemspec

instrumentation/excon/opentelemetry-instrumentation-excon.gemspec

instrumentation/faraday/opentelemetry-instrumentation-faraday.gemspec

instrumentation/graphql/opentelemetry-instrumentation-graphql.gemspec

instrumentation/http/opentelemetry-instrumentation-http.gemspec

instrumentation/http_client/opentelemetry-instrumentation-http_client.gemspec

instrumentation/koala/opentelemetry-instrumentation-koala.gemspec

instrumentation/lmdb/opentelemetry-instrumentation-lmdb.gemspec

instrumentation/mongo/opentelemetry-instrumentation-mongo.gemspec

instrumentation/mysql2/opentelemetry-instrumentation-mysql2.gemspec

instrumentation/net_http/opentelemetry-instrumentation-net_http.gemspec

instrumentation/pg/opentelemetry-instrumentation-pg.gemspec

instrumentation/rack/opentelemetry-instrumentation-rack.gemspec

instrumentation/rails/opentelemetry-instrumentation-rails.gemspec

instrumentation/redis/opentelemetry-instrumentation-redis.gemspec

instrumentation/resque/opentelemetry-instrumentation-resque.gemspec

instrumentation/restclient/opentelemetry-instrumentation-restclient.gemspec

instrumentation/ruby_kafka/opentelemetry-instrumentation-ruby_kafka.gemspec

instrumentation/sidekiq/opentelemetry-instrumentation-sidekiq.gemspec

instrumentation/sinatra/opentelemetry-instrumentation-sinatra.gemspec

propagator/b3/opentelemetry-propagator-b3.gemspec

propagator/jaeger/opentelemetry-propagator-jaeger.gemspec

propagator/ottrace/opentelemetry-propagator-ottrace.gemspec

propagator/xray/opentelemetry-propagator-xray.gemspec

resource_detectors/opentelemetry-resource_detectors.gemspec

sdk/opentelemetry-sdk.gemspec
42:  spec.add_development_dependency 'pry'
43:  spec.add_development_dependency 'pry-byebug'

semantic_conventions/opentelemetry-semantic_conventions.gemspec
```

**Comment by robertlaurin** [comment]
Jruby does not seem to like this.

> bundler: failed to load command: rake (/home/runner/.rubies/jruby-9.2.19.0/bin/rake)
> Bundler::GemNotFound: Could not find gem 'pry-byebug' in rubygems repository https://rubygems.org/ or installed locally.

**Comment by SomalianIvan** [comment]
https://github.com/jruby/jruby/issues/6152#issuecomment-629518669

**Review by robertlaurin — APPROVED** [review]
lgtm, will merge on ci hitting green.

**Review by fbogsany — APPROVED** [review]

**Pull request merged**